modules/terraform/workspace_test.go

```diff
@@ -130,3 +131,99 @@ func TestNameMatchesWorkspace(t *testing.T) {
 		assert.Equal(t, testCase.expected, actual, "Name: %q, Workspace: %q", testCase.name, testCase.workspace)
 	}
 }
+
+func TestWorkspaceDeleteE(t *testing.T) {
+	t.Parallel()
+
+	// state describes an expected status when a given testCase begins
+	type state struct {
+		workspaces []string
+		current    string
+	}
+
+	// testCase describes a named test case with a state, args and expcted results
+	type testCase struct {
+		name              string
+		initialState      state
+		toDeleteWorkspace string
+		expectedCurrent   string
+		expectedError     error
+	}
+
+	testCases := []testCase{
+		{
+			name: "delete another existing workspace and stay on current",
+			initialState: state{
+				workspaces: []string{"staging", "production"},
+				current:    "staging",
+			},
+			toDeleteWorkspace: "production",
+			expectedCurrent:   "staging",
+			expectedError:     nil,
+		},
+		{
+			name: "delete current workspace and switch to a specified",
+			initialState: state{
+				workspaces: []string{"staging", "production"},
+				current:    "production",
+			},
+			toDeleteWorkspace: "production",
+			expectedCurrent:   "default",
+			expectedError:     nil,
+		},
+		{
+			name: "delete a non existing workspace should trigger an error",
+			initialState: state{
+				workspaces: []string{"staging", "production"},
+				current:    "staging",
+			},
+			toDeleteWorkspace: "hellothere",
+			expectedCurrent:   "staging",
+			expectedError:     WorkspaceDoesNotExist("hellothere"),
+		},
+		{
+			name: "delete the default workspace triggers an error",
+			initialState: state{
+				workspaces: []string{"staging", "production"},
+				current:    "staging",
+			},
+			toDeleteWorkspace: "default",
+			expectedCurrent:   "staging",
+			expectedError:     &UnsupportedDefaultWorkspaceDeletion{},
+		},
+	}
+
+	for _, tt := range testCases {
```

Comment:
Two notes:

1. I mention below calling `t.Parallel()`. That means you'll need to store a copy of `tt` here via a silly looking `tt := tt`. See https://github.com/gruntwork-io/terratest/blob/0c883d74782fecaf8e5987a760f5aa9e06a04a7f/modules/aws/ec2_test.go#L52-L54 for an example.
1. NIT: We typically avoid one or two letter variable names, except in very idiomatic cases (e.g., `i` in a for loop or `t` in a Go test case). So `testCase` may be a better name for this variable.